src/datreant/core/limbs.py

```diff
@@ -516,9 +517,15 @@ def add(self, categorydict=None, **categories):
                 if not isinstance(key, string_types):
                     raise TypeError("Keys must be strings.")
 
-                if (isinstance(value, (int, float, string_types, bool))):
+                if isinstance(value, (int, float, string_types, bool)):
                     self._treant._state['categories'][key] = value
-                elif value is not None:
+                elif value is None:
+                    # delete if present
+                    if key in self._treant._state['categories']:
+                        del self._treant._state['categories'][key]
```

Comment:
Maybe a warning might be nice here to notify something potentially bad has happened.